### PR TITLE
[SAIC-726] Improve existing image detection.

### PR DESCRIPTION
**Pull request by antonf**
Improve existing image detection by adding check for tenant_id and
is_public fields.

### CODE REVIEW

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:158: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:239: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:280: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:291: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:294: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:304: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/image/glance_image.py:352: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:358: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:367: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:370: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:390: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:417: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:444: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

File: cloudferrylib/os/image/glance_image.py [hunk 1]
```diff
@@ -450,6 +450,21 @@ def deploy(self, info):
...
+        existing_members = {m.member_id: m
+                            for m in client.image_members.list(image=image_id)}
+        for tenant_name, can_share in image_members.iteritems():
+            tenant_id = self.identity_client.get_tenant_id_by_name(tenant_name)
+            LOG.debug("Deploying image member for image '%s' "
+                      "tenant '%s'", image_id, tenant_name)
+            if (tenant_id in existing_members and
+                    existing_members[tenant_id].can_share != can_share):
+                client.image_members.delete(image_id, tenant_id)
+                del existing_members[tenant_id]
```
Comment (roman-verchikov) [inline cloudferrylib/os/image/glance_image.py]: `del` doesn't look good in most contexts (in this too). Perhaps `filter()` or `set()` of tenant IDs instead?

Comment (antonf) [inline cloudferrylib/os/image/glance_image.py]: Sorry, I don't understand how filter or set can be applied to this case

**Comment by MirantisWorkloadMobility** [comment]
Consider resolving pylint issues: \
``` \
************* Module cloudferrylib.os.image.glance_image
cloudferrylib/os/image/glance_image.py:158: [W0212(protected-access), GlanceImage.get_ref_image] Access to a protected member _resp of a client class
cloudferrylib/os/image/glance_image.py:239: [W0613(unused-argument), GlanceImage.read_info] Unused argument 'kwargs'
cloudferrylib/os/image/glance_image.py:280: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:291: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:294: [W1202(logging-format-interpolation), GlanceImage.make_image_info] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:304: [W0221(arguments-differ), GlanceImage.deploy] Arguments number differs from overridden 'deploy' method
cloudferrylib/os/image/glance_image.py:352: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:358: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:367: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:370: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:390: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:417: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments
cloudferrylib/os/image/glance_image.py:444: [W1202(logging-format-interpolation), GlanceImage.deploy] Use % formatting in logging functions but pass the % parameters as arguments

**Comment by MirantisWorkloadMobility** [comment]
PEP8 check success, Unittests success

**Comment by archyufa** [comment]
+1 Tested in Staging

**Comment by plomakin** [comment]
+1, also fix some of the issue found on devlab

**Comment by roman-verchikov** [comment]
can be merged

**Pull request merged**